Sources/NIO/Socket.swift

```diff
@@ -170,6 +170,33 @@ typealias IOVector = iovec
         }
     }
 
+    func recvmsg(pointer: UnsafeMutableRawBufferPointer, storage: inout sockaddr_storage, storageLen: inout socklen_t, controlBytes: inout UnsafeMutableRawBufferPointer) throws -> IOResult<Int> {
```

Comment:
I don't think we should do `inout UnsafeMutableRawBufferPointer`. That's pretty dangerous because the user can't know that we don't mess with the base address. If we were to mess with the base address, then the user couldn't free this anymore.

The right type for this IMHO is: `inout Slice<UMRBP>`. Then it's clear that the user gives us a slice and we may mess with the slice but the user is always in charge of allocating/freeing the UMRBP and we can't mess with that one.

Reply:
Addressed.